axum/src/docs/routing/with_state.md

```diff
@@ -1,4 +1,4 @@
-Provide the state for the router.
+Provide the state for the router. This methods needs to be called last when building the router, or your code won't compile.
```

Comment:
This is not always true:

```rust
Router::new()
    .with_state(1)
    .route("/", get(|| async {}))
```

That works.

I think instead we should move the example from the state extractor here instead and remove the changes to this sentence.

Reply:
You resolved this conversation but it appears to not be fixed 🤔